engine/src/main/java/io/zeebe/engine/state/appliers/ProcessInstanceElementActivatingApplier.java

```diff
@@ -53,6 +67,21 @@ public void applyState(final long elementInstanceKey, final ProcessInstanceRecor
         Purpose.FAILED);
   }
 
+  private boolean isNonInterruptingEventSubprocess(
+      final ElementInstance flowScopeInstance, final BpmnElementType currentElementType) {
+    final var flowScopeElementType = flowScopeInstance.getValue().getBpmnElementType();
+
+    return currentElementType == BpmnElementType.SUB_PROCESS
+        && flowScopeElementType == BpmnElementType.PROCESS
+        && !isInterrupted(flowScopeInstance);
+  }
+
+  private boolean isInterrupted(final ElementInstance elementInstance) {
+    return elementInstance.getNumberOfActiveTokens() == 2
```

Comment:
What a beautiful magic number :smile: (I know it was already there, no worries)

Reply:
I hate it :laughing: